src/model.cpp

```diff
@@ -373,31 +373,56 @@ void Model::setParameterScale(std::vector<ParameterScaling> const& pscale) {
 std::vector<realtype> const& Model::getParameters() const {
     return originalParameters;
 }
+
+realtype getValueById(std::vector<std::string> ids, std::vector<realtype> values,
+                      std::string const& id, const char* value_name, const char* id_name) {
+    auto it = std::find(ids.begin(), ids.end(), id);
+    if(it != ids.end())
+        return values.at(it - ids.begin());
+    
+    throw AmiException("Could not find %s with specified %s", value_name, id_name);
+}
+    
+void setValueById(std::vector<std::string> ids, std::vector<realtype> values, realtype value,
```

Comment:
This changes only a copy of values (and discards it). Pass ids as `const&`, values as `&`